Kinvey/Kinvey/Client.swift

```diff
@@ -175,14 +180,21 @@ open class Client: NSObject, NSCoding, Credential {
     }
     
     /// Initialize a `Client` instance with all the needed parameters and requires a boolean to encrypt or not any store created using this client instance.
-    open func initialize(appKey: String, appSecret: String, apiHostName: URL = Client.defaultApiHostName, authHostName: URL = Client.defaultAuthHostName, encrypted: Bool, schemaVersion: CUnsignedLongLong = 0, migrationHandler: Migration.MigrationHandler? = nil) {
+    @available(*, deprecated: 3.3.3, message: "Please use initialize(appKey:appSecret:accessGroup:apiHostName:authHostName:encrypted:schema:completionHandler:)")
```

Comment:
so, because we now have an async handler that returns a possible user (using SSO), I thought that would better to concise the schema migration in a tuple with the scheme version and the migration handler

`public typealias Schema = (version: CUnsignedLongLong, migrationHandler: Migration.MigrationHandler?)`